### DISCUSSION TITLE
How to retrieve currentUser through React-Query?

### DISCUSSION THREAD

**Discussion by kchojhu** [TanStack/query#1398] [Q&A]
I keep hearing about React Query could replace server data on Redux/React Context.  For example, say I have a Context (or Redux) that holds the current user information.  I use this 'currentUser' object to show/hide various UI features based on their roles.  So anywhere I need the 'currentUser', I simply write something like below code

`const { currentUser } = useContext(AppContext);`

If I use React Query for that purpose, how would I achieve that? In advance, thanks for your time for answering my question.  This is the only question that's holding me back from using this awesome library!

**Comment by sergiodxa** [comment]
What I do is create a useCurrentUser hook which calls useQuery and made a request to an API endpoint where I could get the currently logged in user, if you use cookies to store your user auth token it will be a simple GET request, if you store your token somewhere else (e.g. localStorage) you can read it inside your query function and then send it to the API.

**Reply by BishwashKumarSah** [reply]
How to do fetch the user data at server side in NextJs?

**Reply by haikali3** [reply]
here is how i would handle it, im sure theres better way as I didnt properly handle if user dont wanna log in from the first place

 ```
const { data, isLoading, isError} = useQuery<UserDetails>({
    queryKey: ["userDetails"],
    queryFn: fetchUserDetails,
    enabled: !isLoggedIn, //run the query if the user is not already logged in
  });

  if (data && !isLoggedIn) {
    setIsLoggedIn(true);
  } else if (isError && isLoggedIn) {
    setIsLoggedIn(false);
  }
```
